### ISSUE TITLE
SequenceMatch: Imitation Learning for Autoregressive Sequence Modelling with Backtracking

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#226]
# TL; DR

- Motivation of GAN + Formulation of MDP + Downstream NLP

# Motivation

- KL divergence can be a poor choice for autoregressive language model training due to its restrictive support and vulnerability to OOD.

# Formulation

<img width="782" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/ec77409f-4f73-40b1-a806-7b0229c9bc4d">

<img width="793" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/3663d32b-caca-4d00-98da-edf1eb87afd8">

- Wish to use some other divergence measure (e.g., chi-squared).
- How to handle the data density without adversarial training?
- Occupancy measure: frequency of (state, action) pair.
- Minimize the f-divergence of occupancy scores (between the data and the model).
- Can be simplified and rendered tractable (derivation presented).
- In practice, prepare partial sequences both from the dataset and model for occupancy measures.
- Add <backspace> operation to support editing, and thus augment the dataset correspondingly (so that it involves backspaces).

<img width="776" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/73a42ec6-4fc5-443f-93b6-36d396351235">

# Evaluation

<img width="772" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/4b700124-29bf-4dc3-8cd1-8f1df240a01a">

<img width="779" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/bb2c59db-b91e-4f06-84a6-1437005d8dd2">

# Inspiration

- Objective-driven AI?
